### PR TITLE
Add support for total_energy_consumption sensor in C3 device.

### PR DESCRIPTION
**Pull request by wasilukm**

### CODE REVIEW

File: doc/C3_hans.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ binary_sensor.{DEVICEID}_zone2_water_temp_mode | binary_sensor | Zone2 Water Tem
 binary_sensor.{DEVICEID}_zone1_room_temp_mode | binary_sensor | Zone1 Room Temperature Mode | 区域1室温模式
 binary_sensor.{DEVICEID}_zone2_room_temp_mode | binary_sensor | Zone2 Room Temperature Mode | 区域2室温模式
 sensor.{DEVICEID}_tank_actual_temperature | sensor | Tank Actual Temperature | 水箱实际温度
+sensor.{DEVICEID}_total_energy_consumption | sensor | Total Energy Consumption | 总能耗。 第一个值可能会延迟，因为更新仅在设备处于活动状态时发送
```
Comment (wasilukm) [inline doc/C3_hans.md:23]: Please correct the translation is wrong. I've used google translate...

File: custom_components/midea_ac_lan/midea/devices/c3/message.py [hunk 2]
```diff
@@ -117,6 +117,13 @@ def __init__(self, body, data_offset=0):
         self.dhw_temp_min = body[data_offset + 20]
         self.tank_actual_temperature = body[data_offset + 21]
 
+class C3Notify1MessageBody(MessageBody):
+    def __init__(self, body, data_offset=0):
+        super().__init__(body)
+        self.total_energy_consumption = (
+            body[data_offset + 4] +
+            (body[data_offset + 3] << 8) +
+            (body[data_offset + 2] << 16))
```
Comment (wasilukm) [inline custom_components/midea_ac_lan/midea/devices/c3/message.py:126]: I'm sure about the two least significant bytes (confirmed with my device) but I assume that at least one more byte is reserved for high values.

**Comment by alecsmro** [comment]
is it working? It will be amazing to gave the consumption stats with this integration <3

**Comment by wasilukm** [comment]
It seems to work fine
obraz

**Comment by alecsmro** [comment]
There is any way to get this energy consumption working, until the integration will be updated with this feature included?
Thanks!

**Comment by alecsmro** [comment]
I managed to get it working! It would be even more exciting to get all the info related to energy consumption, like we've got them inside the MSmartHome App (TBH, IBH, Cooling, Heating, DHW). 

Could be possible?

**Comment by wasilukm** [comment]
@alecsmro Great that it is working for you. If you would like to experiment more you can try https://github.com/wasilukm/midea_ac_lan/tree/hp_state where I have added preliminary support for produced energy, status for heating, DWH, TBH, and IBH.

**Comment by Unwies** [comment]
@wasilukm this looks really great. I tried to get your version into my HA by changing midea_devices.py, device.py and manage.py to your files. But I don't see the power measurements ;( Sorry to bother you this way, but can you tell me, how I get your version active in my HA (sorry, not an expert in GitHub + HA). I deinstalled the old integration, reinstalled, restarted and added the integration. Can you please support me. thx

**Comment by GoodLug** [comment]
I see this thread is still active, with new comments, hopefully you do check it from time to time. Can I reiterate my request that, if it can be done, you add, or point me towards how to add, LWT and RWT (TW_O and TW_I on the wired controller operational parameter pages), plus the water flow (flow rate), comp.current (current draw) and outdoor air temp? They will all contribute to understanding how the heat pump is running.

Again, thanks for your brilliant work, any help very much appreciated.

**Review by Perfect-Web — APPROVED** [review]

**Comment by jioe** [comment]
H, very good job!
 
I have been able to to get the power consumption and generation using your branch.

One question: do you have or do you know where I could get the message memory map information of Midea Heat Pump? I would like to review if there is more interesting information to get from this device. I understand that there is more information of the Heat Bump waiting to be added to the HA :)

Thanks,

**Comment by wasilukm** [comment]
Hi, thank you for all the feedback. I don't know the other values you have mentioned. This all was reverse engineering.
I'm not planning to invest more time in this solution as I'm now using direct Modbus communication, which has documentation for all the data/registers. Not to mention it is more reliable.

**Comment by GoodLug** [comment]
> Hi, thank you for all the feedback. I don't know the other values you have mentioned. This all was reverse engineering. I'm not planning to invest more time in this solution as I'm now using direct Modbus communication, which has documentation for all the data/registers. Not to mention it is more reliable.

Thanks for your reply, interesting. Towards the end of last week I found myself coming to much the same conclusion, that modbus (I'm planning on using a wired connection) is the way to go.

**Comment by georgezhao2010** [comment]
Thank you @wasilukm

**Pull request merged**